### PR TITLE
MM-51907 - Route True-Up Review Telemetry to CWS

### PR DESCRIPTION
**Pull request by ConorMacpherson**
#### Summary

Adds alternative route for true up review submission via CWS in the case that telemetry is disabled on the server.

#### Ticket Link

[MM-51907](https://mattermost.atlassian.net/browse/MM-51907)

#### Test Plan

1. Ensure telemetry is ***DISABLED***.
2. Go to either `System Console > Site Statistics` or `System Console > Team Statistics`
3. Submit the true-up review by clicking the button in the true-up review panel.

To see the true-up review panels outside of true-up season, you should be able to modify records in the table `trueupreviewhistory` within Mattermost's database to a date within the next few days.

#### Related PRs

[CWS](https://github.com/mattermost/customer-web-server/pull/1163)
[Enterprise](https://github.com/mattermost/enterprise/pull/1417)

#### Release Note

```release-note
NONE
```

[MM-51907]: https://mattermost.atlassian.net/browse/MM-51907?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/channels/src/components/analytics/true_up_review.tsx [hunk 1]
```diff
@@ -223,10 +223,6 @@ const TrueUpReview: React.FC = () => {
         return null;
     }
 
-    if (telemetryEnabled) {
-        return null;
-    }
-
```
Comment (marianunez) [inline webapp/channels/src/components/analytics/true_up_review.tsx:229]: If telemetry is enabled we still don't want to show the button to submit the True up review correct? When telemetry is enabled we already have the data that we need on our side for true up verifications

Comment (ConorMacpherson) [reply webapp/channels/src/components/analytics/true_up_review.tsx:229]: Ah, okay. So should this be a recurring check on a schedule in the backend to just automatically send the true-up data off to CWS?

File: server/channels/api4/license.go [hunk 2]
```diff
@@ -354,15 +354,22 @@ func requestTrueUpReview(c *Context, w http.ResponseWriter, r *http.Request) {
 	// Do not send true-up review data if the user has already requested one for the quarter.
 	// And only send a true-up review via as a one-time telemetry request if telemetry is disabled.
 	telemetryEnabled := c.App.Config().LogSettings.EnableDiagnostics
-	if telemetryEnabled != nil && !*telemetryEnabled {
+	if telemetryEnabled != nil && *telemetryEnabled {
 		// Send telemetry data
 		c.App.Srv().GetTelemetryService().SendTelemetry(model.TrueUpReviewTelemetryName, profileMap)
```
[the remark this answers is not in the document]
Comment (ConorMacpherson) [reply server/channels/api4/license.go:359]: Should all telemetry just be routed to CWS?

File: webapp/channels/src/components/analytics/true_up_review.tsx (hunk 1 above)
Comment (ConorMacpherson) [reply webapp/channels/src/components/analytics/true_up_review.tsx:229]: Ah, my mistake. I thought this was negated. For some reason, I had thought that the feature was only enabled when telemetry is enabled.

Comment (marianunez) [reply webapp/channels/src/components/analytics/true_up_review.tsx:229]: > So should this be a recurring check on a schedule in the backend to just automatically send the true-up data off to CWS?

We still want the button so they explicitly share the data with us but we just need to route it through the portal given that telemetry is disabled in their server

**Review by nickmisasi — APPROVED** [review]

File: server/channels/api4/license.go [hunk 3]
```diff
@@ -351,25 +351,26 @@ func requestTrueUpReview(c *Context, w http.ResponseWriter, r *http.Request) {
...
-	if telemetryEnabled != nil && !*telemetryEnabled {
-		// Send telemetry data
-		c.App.Srv().GetTelemetryService().SendTelemetry(model.TrueUpReviewTelemetryName, profileMap)
-
-		// Update the review status to reflect the completion.
-		status.Completed = true
-		c.App.Srv().Store().TrueUpReview().Update(status)
+	// True-up is only enabled when telemetry is disabled. When telemetry is enabled, we already have all the data necessary
+	// for true-up reviews to be completed.
+	err = c.App.Cloud().SubmitTrueUpReview(c.AppContext.Session().UserId, profileMap)
```
Comment (neallred) [inline server/channels/api4/license.go:356]: Should we still check for whether telemetry is enabled for completeness' sake?

**Review by neallred — APPROVED** [review]
LGTM! It would maybe still be good to check if telemetry is enabled or not, even though I'm guessing we shouldn't get requests to this endpoint if it is off.

File: server/channels/api4/license.go (hunk 3 above)
Comment (ConorMacpherson) [reply server/channels/api4/license.go:356]: Better safe than sorry! I'll add the check :)

**Comment by amyblais** [comment]
@ConorMacpherson Please help open manual cherry-pick PRs for ``release-7.9`` and ``release-7.10`` branches.

**Comment by amyblais** [comment]
@ConorMacpherson And also to the ``cloud`` branch.

**Comment by amyblais** [comment]
@ConorMacpherson I have merge permissions for ``cloud`` branch but not for other branches. I see some CI failures in the cherry-pick PRs. Do they look safe to merge despite of the CI failures?

**Comment by ConorMacpherson** [comment]
> Do you know what the cause of the failure is and how to fix it?

My mistake, I thought the most recent commit to the `cloud` branch of the enterprise was failing with the same issue, but it's not. It's two unrelated failures. I'll see if I can fix the issues on my cherry-pick.

**Comment by amyblais** [comment]
This and the corresponding enterprise PR (https://github.com/mattermost/enterprise/pull/1417) can be cherry-picked to the ``cloud-latest`` branch. A few recent cherry-picks got removed from the cloud branch due to the ongoing incident.

**Pull request merged**